easypoll/src/main/java/com/eis0/easypoll/poll/BinaryPoll.java

```diff
@@ -0,0 +1,300 @@
+     * @param users The {@link Network} representing users subscribed to the poll.
+     * @throws IllegalArgumentException When one of the arguments is empty or different authors are given.
+     * @author Matteo Carnelos
+     */
+    public BinaryPoll(long number, @NonNull String name, @NonNull String question,
+                      @NonNull Network author, @NonNull Network users) {
+        if(name.isEmpty()) throw new IllegalArgumentException("Can't create poll with empty name.");
+        if(question.isEmpty()) throw new IllegalArgumentException("Can't create poll with empty question.");
+        if(author.size() > 1) throw new IllegalArgumentException("There can only be one poll author.");
+        if(users.isLocalNetwork()) throw new IllegalArgumentException("A poll must contain at least one user.");
+        this.number = number;
+        this.name = name;
+        this.question = question;
+        this.author = author;
+        this.users = users;
+        incoming = true;
+    }
+
+    /**
+     * Create a new poll from this device.<br>
+     * The poll number is calculated from the counter stored in the internal memory of the device.
+     * The author's {@link Network} is the local network, that is the one composed only with the
+     * local host, see {@link Network} for more details.
+     *
+     * @param name The name of the poll.
+     * @param question The question to ask all users.
+     * @param users Users to include in the poll.
+     * @throws IllegalArgumentException If one of the arguments is empty.
+     * @author Matteo Carnelos
+     */
+    public BinaryPoll(@NonNull String name, @NonNull String question, @NonNull Network users) {
+        this(pollsCount+1, name, question, NetworksPool.obtainLocalNetwork(), users);
+        incoming = false;
+        pollsCount++;
+        savePollsCountToInternal();
+    }
+
+    // ---------------------------- DATA STORING ---------------------------- //
+
+    /**
+     * Set the SharedPreferences reference to which save the {@link #pollsCount} value.
+     *
+     * @author Matteo Carnelos
+     */
+    public static void setSharedPreferences(SharedPreferences sharedPreferences) {
+        mSharedPreferences = sharedPreferences;
+    }
+
+    /**
+     * Save the polls counter (for ids) to the internal storage.
+     *
+     * @author Matteo Carnelos
+     */
+    private static void savePollsCountToInternal() {
+        if(mSharedPreferences == null) return;
+        SharedPreferences.Editor editor = mSharedPreferences.edit();
+        editor.putLong(POLLS_COUNT_KEY, pollsCount);
+        editor.apply();
+    }
+
+    /**
+     * Load the polls counter (for ids) from the internal storage. If none is found, load the default
+     * one (0).
+     *
+     * @author Matteo Carnelos
+     */
+    public static void loadPollsCountFromInternal() {
+        pollsCount = mSharedPreferences.getLong(POLLS_COUNT_KEY, 0);
+    }
```

Comment:
> I'm not sure about the reason why you are enabling mSharedPreferences to be null.

I did that because the shared preferences pointer is obtained from the context of the application and this class doesn't have any reference to that, so two was the possible ways:
1. **Asking the context (or, better, directly the shared preferences pointer) in the constructor of the class.**
This solution is usually adopted when the resource needed is indispensable for the correct working of the class or it is needed during the creation of the object. However, in my case the storing feature is not indispensable, as the class will still work without it (obviously losing the storing feature if the user doesn't apply any other type of data retention), and it is not needed during creation. So I went for solution 2.
2. **Initially setting the shared preferences pointer null and letting the user set its value if needed.**
As I said before, the storing feature is meant to be optional, so this solution was the right one for my case.

> In the savePollsCountToInternal() if the value is null you simply don't do anything, not even an advertisement for the programmer

I chose to don't do anything because launching an exception was too much in my opinion, I usually launch exceptions when the working of the program is compromised and it can't go on, and this wasn't the case.
You're totally right about the advertisement for the programmer, I missed that. It could be done by informing the user in the specifics of the method that, if `null`, the method will terminate quietly. That's what I'll do, thanks!

> I assume that if the mSharedPreferences is null a NullPointerException will be thrown.

That's my fault, I missed the nullity check like I do in `savePollsCountToInternal()`. Will do!

Reply:
Exceptional explanation, I had the impression that there was a reason for your choice considering the rest of the code; anyway I think that's all for this code review.

I really like your style, how you split the different methods (Getters, Setters etc...), the names of the variables you use, maybe you could add some more explicative descriptions for methods and classes, but I probably say it because of my attitude to writing poems for the simplest commands. It surely was easier for me understanding what you did considering that I already knew the purpose of those classes, but the order in the code helped a lot.

The most interesting part of the code is, in my opinion, the one which doesn't operate directly with the UI, because even if it's made perfectly it's like a showcase of everything you can do in Android to create an Interface; still really helpful considering that right now I'm totally unable to do something similar.

Reply:
I really appreciate that, thanks a lot!😁